src/main/flight/rth_estimator.c

```diff
@@ -153,7 +153,7 @@ static float calculateRemainingEnergyBeforeRTH(bool takeWindIntoAccount) {
           && feature(FEATURE_CURRENT_METER) && (batteryMetersConfig()->cruise_power > 0)
           && (currentBatteryProfile->capacity.unit == BAT_CAPACITY_UNIT_MWH) && (currentBatteryProfile->capacity.value > 0)
           && navigationPositionEstimateIsHealthy() && isImuHeadingValid() && (navConfig()->fw.cruise_speed > 0)
-          && ((!STATE(FIXED_WING_LEGACY)) || !isNavLaunchEnabled() || (isNavLaunchEnabled() && isFixedWingLaunchDetected()))
+          && ((!STATE(FIXED_WING_LEGACY)) || !isNavLaunchEnabled() || (isNavLaunchEnabled() && (isFixedWingLaunchDetected() || isFixedWingLaunchFinishedOrAborted())))
```

Comment:
This needs clean-up. This whole logic condition is impossible to read.

Reply:
I agree, not not's are really confusing to get your head around. I'm also not sure why the launch stuff is in here anyway.  RTH estimator originally only worked if you used Nav Launch but it was changed to work without when I raised this as a problem if you didn't use Launch, and it still works fine. Does the short time between arming and launching really make enough of a difference to the estimation to require launch to be considered ?

Reply:
Is this clearer ?

```
static float calculateRemainingEnergyBeforeRTH(bool takeWindIntoAccount) {
    // Fixed wing only for now
    if (!(STATE(FIXED_WING_LEGACY) || ARMING_FLAG(ARMED))) {
        return -1;
    }
    
#ifdef USE_WIND_ESTIMATOR
    if !(takeWindIntoAccount && isEstimatedWindSpeedValid()) {
        return -1;
    }
#endif

    const bool batterySettingsAreOK = feature(FEATURE_VBAT) && feature(FEATURE_CURRENT_METER) && batteryWasFullWhenPluggedIn();
    const bool rthEstimatorSettingsAreOK = batteryMetersConfig()->cruise_power > 0 && currentBatteryProfile->capacity.unit == BAT_CAPACITY_UNIT_MWH &&currentBatteryProfile->capacity.value > 0 && navConfig()->fw.cruise_speed > 0; 
    const bool navigationIsOK = navigationPositionEstimateIsHealthy() && isImuHeadingValid();
    const bool launchIsActiveOrFinished = isNavLaunchEnabled() && (isFixedWingLaunchDetected() || isFixedWingLaunchFinishedOrAborted());
    
    if (!(batterySettingsAreOK && rthEstimatorSettingsAreOK && navigationIsOK && launchIsActiveOrFinished) {
        return -1;
    }
```

Reply:
Thinking this code would be better placed in calculateRemainingDistanceBeforeRTH which calls calculateRemainingEnergyBeforeRTH via another function. Why call 2 functions before checking if the criteria for it to work are met when you could check at the initial calling function and stop there if not ?

Reply:
Realised launchIsActiveOrFinished is wrong, should be the following instead:

`const bool launchIsInactiveOrDetected = !isNavLaunchEnabled() || (isFixedWingLaunchDetected() || isFixedWingLaunchFinishedOrAborted());`

Or just delete completely since it doesn't seem necessary anyway.